src/main/java/com/github/reducktion/socrates/validator/CanadaIdValidator.java

```diff
@@ -0,0 +1,66 @@
+package com.github.reducktion.socrates.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * National Identification Number validator for Canada.
+ *
+ * A wikipedia source can be found in: https://en.wikipedia.org/wiki/Social_Insurance_Number
+ *
+ **/
+class CanadaIdValidator implements IdValidator {
+
+    private static final int ID_NUMBER_OF_CHARACTERS = 9;
+
+    @Override
+    public boolean validate(final String id) {
+        if (id == null) {
+            return false;
+        }
+
+        final String sanitizedId = sanitize(id);
+
+        if (sanitizedId.length() != ID_NUMBER_OF_CHARACTERS
+            || !StringUtils.isNumeric(sanitizedId)
+        ) {
+            return false;
+        }
+
+        return validateSequence(sanitizedId);
+    }
+
+    private String sanitize(final String id) {
+        return id
+            .trim()
```

Comment:
You don't need the `trim()` here because you will replace all the spaces in the next line.